processor/processor.go

```diff
@@ -1372,6 +1372,107 @@ func getDiffMetrics(inPU, pu string, inCountMetadataMap map[string]MetricMetadat
 	return diffMetrics
 }
 
+func generateFilteredAndAddedMetrics(inPU, pu string, messageIds []string, messageIdKeyMap map[string]string, inCountMetadataMap map[string]MetricMetadata, outputEvents []transformer.TransformerEvent, failedEvents []transformer.TransformerResponse) ([]*types.PUReportedMetric, []*types.PUReportedMetric) {
```

Comment:
A common fallacy in processor right now is the assumption that messageIds in a batch are unique. This doesn't stand true if deduplication is not enabled.

Reply:
@atzoum @Sidddddarth any suggestions around what would be the unique composite key - that is present in both input and output metadata of transformer. should we do `messageId + jobId` and add jobIds similar to messageIds in transformer response for transformBatch case.